emily/handler/src/api/handlers/deposit.rs

```diff
@@ -513,11 +503,23 @@ async fn update_deposits(
         body.try_into_validated_update_request(api_state.chaintip().into())?;
 
     // Create aggregator.
-    let mut updated_deposits: Vec<(usize, Deposit)> =
+    let mut updated_deposits: Vec<(usize, DepositWithStatus)> =
         Vec::with_capacity(validated_request.deposits.len());
 
     // Loop through all updates and execute.
     for (index, update) in validated_request.deposits {
+        if update.is_err() {
+            updated_deposits.push((
+                index,
+                DepositWithStatus {
+                    deposit: Deposit::default(),
+                    status: StatusCode::BAD_REQUEST.as_u16(),
+                },
+            ));
+            continue;
+        }
+        let update = update.unwrap();
```

Comment:
```suggestion
        let Ok(update) = update else {
            updated_deposits.push((
                index,
                DepositWithStatus {
                    deposit: Deposit::default(),
                    status: StatusCode::BAD_REQUEST.as_u16(),
                },
            ));
            continue;
        }
```
Maybe for the future, this already has two approvals